borgmatic/hooks/mysql.py

```diff
@@ -225,31 +236,69 @@ def restore_data_source_dump(
     mysql_restore_command = tuple(
         shlex.quote(part) for part in shlex.split(data_source.get('mysql_command') or 'mysql')
     )
-    restore_command = (
-        mysql_restore_command
-        + ('--batch',)
-        + (
-            tuple(data_source['restore_options'].split(' '))
-            if 'restore_options' in data_source
-            else ()
+
+    dump_format = data_source.get('format')
+    is_directory_format = dump_format == 'directory'
+
+    if is_directory_format:
+        mysql_restore_command = tuple(
+            shlex.quote(part) for part in shlex.split(data_source.get('mysql_command') or 'mysql')
         )
-        + (('--host', hostname) if hostname else ())
-        + (('--port', str(port)) if port else ())
-        + (('--protocol', 'tcp') if hostname or port else ())
-        + (('--user', username) if username else ())
-    )
+        restore_command = (
+            mysql_restore_command
+            + ('--batch',)
+            + (
+                tuple(data_source['restore_options'].split(' '))
+                if 'restore_options' in data_source
+                else ()
+            )
+            + (('--host', hostname) if hostname else ())
+            + (('--port', str(port)) if port else ())
+            + (('--protocol', 'tcp') if hostname or port else ())
+            + (('--user', username) if username else ())
+        )
+    else:
+        restore_command = (
+            mysql_restore_command
+            + ('--batch',)
+            + (
+                tuple(data_source['restore_options'].split(' '))
+                if 'restore_options' in data_source
+                else ()
+            )
+            + (('--host', hostname) if hostname else ())
+            + (('--port', str(port)) if port else ())
+            + (('--protocol', 'tcp') if hostname or port else ())
+            + (('--user', username) if username else ())
+        )
```

Comment:
I think I'm missing something.. These two commands look identical to me. Maybe it's just still WIP.